uring/src/main/scala/fs2/io/uring/net/UringDatagramSocket.scala

```diff
@@ -117,9 +114,24 @@ private[net] object UringDatagramSocket {
   def apply[F[_]](ring: Uring[F], fd: Int)(implicit
       F: Async[F]
   ): Resource[F, UringDatagramSocket[F]] =
-    ResizableBuffer(65535).evalMap { buf =>
-      (Mutex[F], Mutex[F]).mapN(new UringDatagramSocket(ring, fd, buf, 65535, _, _))
-    }
+    ResizableBuffer(65535)
+      .evalMap { buf =>
+        (Mutex[F], Mutex[F]).mapN { (readMutex, writeMutex) =>
+          val iov = malloc(sizeof[uio.iovec]).asInstanceOf[Ptr[uio.iovec]]
+          val remoteAddr = malloc(sizeof[sockaddr_storage]).asInstanceOf[Ptr[sockaddr_storage]]
+          val msg = malloc(sizeof[msghdr]).asInstanceOf[Ptr[msghdr]]
+
+          new UringDatagramSocket(ring, fd, buf, 65535, readMutex, writeMutex, iov, msg, remoteAddr)
+        }
+      }
+      .flatMap { socket =>
+        val release: F[Unit] = F.delay {
+          free(socket.iov.asInstanceOf[Ptr[Byte]])
+          free(socket.msg.asInstanceOf[Ptr[Byte]])
+          free(socket.remoteAddr.asInstanceOf[Ptr[Byte]])
+        }
+        Resource.make(F.pure(socket))(_ => release)
+      }
```

Comment:
Instead of `Resource.eval(...).flatMap(Resource.make(pure(...))(...))` we should just write `Resource.make(...)(...)` 🙂